### PR TITLE
Use v2.0 code with alpine-based python

### PR DESCRIPTION
**Pull request by natarajaya**
We currently use `v2.0` image that is debian-based and now has [382 vulnerabilities](https://console.cloud.google.com/gcr/images/gpii-gcp-dev-natarajaya/GLOBAL/k8s-snapshots@sha256:09e9320fdacaebdfbec98394f9012ea3de9f035435598bf9d642340349833eba/details?tab=vulnz&project=gpii-gcp-dev-natarajaya&authuser=1&organizationId=247149361674&gcrVulnzListsize=30), including 2 critical.

I created new branch from `v2.0` tag and updated `Dockerfile` to use alpine-based image instead, that [does not have vulnerabilities](https://console.cloud.google.com/gcr/images/gpii-gcp-dev-natarajaya/GLOBAL/k8s-snapshots@sha256:574f55adf45913eef883bbd6e23f091193c96dfe662bd52e2cb9bc7135c9be40/details?tab=vulnz&project=gpii-gcp-dev-natarajaya&authuser=1&organizationId=247149361674).

This PR also contains stuff needed for CI to build images.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,7 +1,17 @@
...
+        make \
+    && pip install -r requirements.txt \
+    && python setup.py install \
+    && apk del \
+        g++ \
+        make
+
+ENV TZ UTC
+
 CMD ["k8s-snapshots"]
```
Comment (stepanstipl) [inline Dockerfile:20]: Is running as `root` required for some reason?

**Review by stepanstipl — APPROVED** [review]
LGTM. Do you plan to open a PR against upstream?

**Comment by natarajaya** [comment]
@stepanstipl Thanks for the review!

>  Do you plan to open a PR against upstream?

Not sure. Upstream does not look entirely dead, but not much is happening there. Since this addresses security issues with the only stable release (latest build does not work properly with GKE), I think these fixes make sense.

File: Dockerfile (hunk 1 above)
Comment (natarajaya) [reply Dockerfile:20]: I don't think so, but need to test. Upstream image runs as root.

Comment (amatas) [reply Dockerfile:20]: The command runs as `nobody:nobody` specified in the line 18, am I right?

**Comment by amatas** [comment]
Thanks @natarajaya , have you tested this code during enough time to make sure that it does not have the weird behavior of the other version?

File: Dockerfile (hunk 1 above)
Comment (natarajaya) [reply Dockerfile:20]: @amatas Yes.

**Comment by natarajaya** [comment]
@amatas Thanks for the review!

You can check the [snapshots metric](https://app.google.stackdriver.com/metrics-explorer?project=gpii-gcp-dev-natarajaya&timeSelection=%7B%22timeRange%22:%221w%22%7D&xyChart=%7B%22dataSets%22:%5B%7B%22timeSeriesFilter%22:%7B%22filter%22:%22metric.type%3D%5C%22logging.googleapis.com%2Fuser%2Fcompute.disks.createSnapshot%5C%22%20resource.type%3D%5C%22gce_disk%5C%22%22,%22perSeriesAligner%22:%22ALIGN_SUM%22,%22crossSeriesReducer%22:%22REDUCE_SUM%22,%22secondaryCrossSeriesReducer%22:%22REDUCE_NONE%22,%22minAlignmentPeriod%22:%22300s%22,%22groupByFields%22:%5B%22resource.label.%5C%22zone%5C%22%22%5D,%22unitOverride%22:%221%22%7D,%22targetAxis%22:%22Y1%22,%22plotType%22:%22LINE%22%7D%5D,%22options%22:%7B%22mode%22:%22COLOR%22%7D,%22constantLines%22:%5B%5D,%22timeshiftDuration%22:%220s%22,%22y1Axis%22:%7B%22label%22:%22y1Axis%22,%22scale%22:%22LINEAR%22%7D%7D&isAutoRefresh=true) in my dev env, I don't see any issues with it and this version is deployed for almost a week.

**Pull request merged**